### PR TITLE
try to get CI working

### PR DESCRIPTION
**Pull request by bendichter**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

# Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would break back-compatibility)
- [ ] Changes in documentation or new examples for demos and/or use cases.

# Branching
- All PRs should be made against the **dev** branch. The master branch is not often merged back to dev.
- If you want to get your PR out to the world faster (urgent bugfix), poke pgunn to cut a release; this will get it onto github and into conda faster

# Has your PR been tested?

If you're fixing a bug or introducing a new feature it is recommended you run the tests by typing

```python caimanmanager.py test```

and

```python caimanmanager.py demotest```

prior to submitting your pull request. 

Please describe any additional tests that you ran to verify your changes. If they are fast you can also
include them in the folder 'caiman/tests/` and name them `test_***.py` so they can be included in our lists of tests.

### CODE REVIEW

**Comment by pgunn** [comment]
Hello,
Can you test this without landing it? This has some changes that I'd prefer not land in the dev branch:
1) long-term I don't want to add dependencies on the pip install to anything as if we ever find a good way to do so it'd be great to end support for it
2) I don't really want reformatting of setup.py (why did that happen?)
3) caimanmanager has a test subcommand that's preferable

**Comment by pgunn** [comment]
Oh, I also noticed that this uses github actions; we've also run into issues with this (random people keep gobbling up our quota)

I think to get CI working again we'll need to think through some of these things more; right now this is premature. It's a worthy thought though.

**Comment by bendichter** [comment]
@pgunn first of all, sorry for springing this PR on you without creating an Issue first. I was testing out some ideas and couldn't find a way to test my GitHub Actions workflow without opening a PR. I now have one running on my personal fork of CaImAn, so this one isn't critical.

 I also wanted to respond to some of your questions and comments:

> I also noticed that this uses github actions; we've also run into issues with this (random people keep gobbling up our quota)

My team has had a lot of success with GitHub Actions on our projects. It's quite easy to set up a free testing matrix across OS and Python versions, and to automatically calculate and publish test coverage. For a public repo like CaImAn, my understanding is [you have an unlimited number of minutes](https://github.com/orgs/community/discussions/26054), so there really should be no worry about users using worker time.

> long-term I don't want to add dependencies on the pip install to anything as if we ever find a good way to do so it'd be great to end support for it

OK, I can try a conda install

> I don't really want reformatting of setup.py (why did that happen?)

I was fixing some inconsistent indenting that my editor found. I can undo this.

**Comment by pgunn** [comment]
Regarding the unlimited number of minutes for public repos, I'm surprised, but if true this is great. I'll have to dig through our org logs to see if that lines up with our experience (meaning - is this still policy?) but if it is, that substantially opens the door for us for options.

Don't worry about the not opening an issue first if you don't mind the diff potentially not making it in. It's sometimes less hassle to JFDI so long as that understanding is there.

Consider the "caimanmanager test" command.

If you:
* can get this working with conda (and revert the other parts related to requirements.txt)
* revert the style changes
* do "caimanmanager.test" instead of manually running the tests
* and if all that works

then so long as this doesn't consume or get impacted by our org-level "minutes allowance", I'm up for merging.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          
+          
+          #   - name: install caiman
+          #     run: |
+          #       mamba install -c conda-forge caiman
+
+      - name: install caiman
+        run: |
+          mamba env update -n base --file environment.yml
+          mamba install Cython -y
```
Comment (pgunn) [inline .github/workflows/ci.yml:41]: environment.yml covers the cython install

Comment (bendichter) [reply .github/workflows/ci.yml:41]: Hmm I am getting 

```pytb
ERROR: Failure: ImportError (cannot import name 'oasis' from partially initialized module 'caiman.source_extraction.cnmf' (most likely due to a circular import) (/home/runner/work/CaImAn/CaImAn/caiman/source_extraction/cnmf/__init__.py))
```

which indicates that oasis isn't being compiled correctly on `pip install`. This is I think the last critical piece. Any tips?

**Comment by pgunn** [comment]
chdir out of the source tree before trying to do anything with caiman and you'll probably have better luck; the way python import paths work has a complex interaction with builds.

**Comment by pgunn** [comment]
For the "Install Caiman" section, maybe try this:

          mamba env update -n base --file environment.yml
          pip install .
          cd /tmp
          caimanmanager install

For the "Test" section, maybe try this:
         cd /tmp
         caimanmanager test

**Comment by bendichter** [comment]
It looks like it might a compilation issue, since `python setup.py build_ext --inplace` fails

```pytb
linux-gnu.so -> caiman/source_extraction/cnmf
Traceback (most recent call last):
  File "/home/runner/work/CaImAn/CaImAn/setup.py", line 66, in <module>
    setup(
  File "/usr/share/miniconda/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/share/miniconda/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/usr/share/miniconda/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
    self.run_command(cmd)
  File "/usr/share/miniconda/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
    super().run_command(command)
  File "/usr/share/miniconda/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/usr/share/miniconda/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 87, in run
    self.copy_extensions_to_source()
  File "/usr/share/miniconda/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 110, in copy_extensions_to_source
    if ext._needs_stub:
       ^^^^^^^^^^^^^^^
AttributeError: 'Extension' object has no attribute '_needs_stub'
```

**Comment by pgunn** [comment]
That's with the "install a new pip over conda's pip" stuff included?
That step is definitely not a good idea

**Comment by bendichter** [comment]
That was inspired by [this issue](https://github.com/grpc/grpc/issues/8004), which seemed to suggest that the error might be due to out-of-date setup tools, but yeah, it didn't work. I'm not able to continue debugging this right now, but I'll come back to it later

**Comment by bendichter** [comment]
@pgunn if you want to follow along, I have the CI running [here](https://github.com/bendichter/CaImAn/pull/1), and that is where the build outputs are being generated.

**Comment by EricThomson** [comment]
@bendichter  as I mentioned I got that same error when doing a dev install a couple of weeks ago, and iirc the only way I got around it was some cargo-culty thing like `pip install . -vvv` and I don't even want to know why the three verbose flags helped or if that was placebo. It may have just been removing the `-e`. This was in my local `dev` branch, but on Windows. It was right before the workshop and I had to just get things working quickly so I didn't have time to think/delve too deeply.

**Comment by bendichter** [comment]
@EricThomson thanks for the hint. I ended up needing to add -e and that seems to have fixed it.

**Comment by bendichter** [comment]
@EricThomson thanks for the hint. I ended up needing to add -e and that seems to have fixed it.

**Comment by bendichter** [comment]
superseded by https://github.com/flatironinstitute/CaImAn/pull/1228

**Pull request closed without merging**